File: adapters/utility-adapter/go/adapter/adapter.go
package adapter
import "context"
type EFNUtilityAdapter interface {
    ValidateCustomer(ctx context.Context, req ValidateRequest) (*ValidateResponse, error)
    DispenseValue(ctx context.Context, req DispenseRequest) (*DispenseResponse, error)
    TransactionStatus(ctx context.Context, efnReference string) (*TransactionStatusResponse, error)
}

File: adapters/utility-adapter/python/efn_utility_adapter/adapter.py
import abc
from .models import *

class EFNUtilityAdapter(abc.ABC):
    @abc.abstractmethod
    def validate_customer(self, req: ValidateRequest) -> ValidateResponse:
        pass

    @abc.abstractmethod
    def dispense_value(self, req: DispenseRequest) -> DispenseResponse:
        pass

    @abc.abstractmethod
    def transaction_status(self, efn_reference: str) -> TransactionStatusResponse:
        pass
